core/autocrypt/crypto.py

```diff
@@ -0,0 +1,574 @@
+        logger.debug('secretppgpykeys %s', self.secretpgpykeys)
+
+    def gen_secret_key(self,
+                       emailadr='alice@testsuite.autocrypt.org',
+                       alg_key=PubKeyAlgorithm.RSAEncryptOrSign,
+                       alg_subkey=PubKeyAlgorithm.RSAEncryptOrSign,
+                       size=2048,
+                       add_subkey=True,
+                       protected=False,
+                       _own=True):
+        # RSAEncrypt is deprecated, therefore using RSAEncryptOrSign
+        # also for the subkey
+        """Generate PGPKey object.
+
+        :param alg_key: algorithm for primary key
+        :param alg_subkey: algorithm for subkey
+        :param size: key size
+        :param emailadr: e-mail address
+        :return: key
+        :type alg_key: PubKeyAlgorithm
+        :type alg_subkey: PubKeyAlgorithm
+        :type size: integer
+        :type emailadr: string
+        :rtype: PGPKey
+
+        """
+        # NOTE: default algorithm was decided to be RSA and size 2048.
+        skey = PGPKey.new(alg_key, size)
+        logger.debug('new pgpkey')
+        # NOTE: pgpy implements separate attributes for name and e-mail
+        # address. Name is mandatory.
+        # Here e-mail address is used for the attribute name .
+        # If name attribute would be set to empty string
+        # and email to the e-mail address, the uid would be
+        # ' <e-mail address>', for instance:
+        # " <alice@testsuite.autocrypt.org>" - which we do not want.
+        uid = PGPUID.new(emailadr)
+        logger.debug('new uid %s', uid)
+        # NOTE: it is needed to specify all arguments in current pgpy
+        # version.
+        # FIXME: see which defaults we would like here
+        skey.add_uid(
+                uid,
+                usage={KeyFlags.Sign},
+                hashes=[HashAlgorithm.SHA512, HashAlgorithm.SHA256],
+                ciphers=[SymmetricKeyAlgorithm.AES256,
+                         SymmetricKeyAlgorithm.AES192,
+                         SymmetricKeyAlgorithm.AES128],
+                compression=[CompressionAlgorithm.ZLIB,
+                             CompressionAlgorithm.BZ2,
+                             CompressionAlgorithm.ZIP,
+                             CompressionAlgorithm.Uncompressed])
+        logger.debug('uid added')
+        if add_subkey is True:
+            subkey = PGPKey.new(alg_subkey, size)
+            logger.debug('new subkey')
+            skey.add_subkey(subkey,
+                            usage={KeyFlags.EncryptCommunications,
+                                   KeyFlags.EncryptStorage})
+            logger.debug('subkey added')
+        if protected is True:
+            passphrase = getpass.getpass()
+            skey.protect(
+                             passphrase,
+                             SymmetricKeyAlgorithm.AES256,
+                             HashAlgorithm.SHA256)
+            logger.debug('Key protected')
```

Comment:
currently py-autocrypt does not support passphrases by design and if we added passphrases we cannot just call into interactive "getpass()" here.  So better remove the support for interactive input here.